section3/CompoundPractice/contracts/AaveFlashLoan.sol

```diff
@@ -0,0 +1,99 @@
+pragma solidity ^0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IPool} from "./IPool.sol";
+import {IPoolAddressesProvider} from "./IPoolAddressesProvider.sol";
+import {IFlashLoanSimpleReceiver} from "./IFlashLoanSimpleReceiver.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "compound-protocol/contracts/CErc20.sol";
+import "compound-protocol/contracts/CTokenInterfaces.sol";
+
+contract AaveFlashLoan is IFlashLoanSimpleReceiver {
+    struct CallbackData {
+        address USDC;
+        address UNI;
+        address cUSDC;
+        address cUNI;
+        address borrower;
+        uint256 liquidateAmount;
+        address liquidator;
+    }
+
+    //address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant POOL_ADDRESSES_PROVIDER =
+        0x2f39d218133AFaB8F2B819B1066c7E434Ad94E9e;
+
+    function execute(bytes calldata callbackData) external {
+        CallbackData memory data = abi.decode(callbackData, (CallbackData));
+        POOL().flashLoanSimple(
+            address(this),
+            data.USDC,
+            data.liquidateAmount,
+            callbackData,
+            0
+        );
+        IERC20(data.USDC).transfer(
+            data.liquidator,
+            IERC20(data.USDC).balanceOf(address(this))
+        );
+    }
+
+    function executeOperation(
```

Comment:
Almost perfect, 2 checks are missed here:
1. msg.sender needs to be the aave pool
2. initiator needs to be your own address